File: exchange-of-values/swap.c
#include <stdio.h>

void swap(int *x, int *y)
{
    int temp;
    temp = *x; *x = *y; *y = temp;
}

int main()
{
    int a = 2;
    int b = 3;
    printf("Before: a = %d, b = %d\n", a, b);
    swap(&a, &b);
    printf("After: a = %d, b = %d\n", a, b);
}

File: exchange-of-values/swap.py
def swap(x, y):
    temp = x
    x = y
    y = temp
    return (x, y)

a, b = 3, 2
print("Before: a = %d, b = %d" % (a, b))
a, b = swap(a, b)
print("After: a = %d, b = %d" % (a, b))
